File: cwt-spring/src/main/java/com/cwtsite/cwt/entity/Comment.java
package com.cwtsite.cwt.entity;

import com.cwtsite.cwt.domain.game.entity.Game;
import com.cwtsite.cwt.domain.user.repository.entity.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.sql.Timestamp;
import java.util.Objects;

/**
 * A Comment.
 */
@Entity
@Table(name = "comment")
@SequenceGenerator(name = "comment_seq", sequenceName = "comment_id_seq")
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "comment_seq")
    private Long id;

    @NotNull
    @Column(name = "body", nullable = false, columnDefinition = "text")
    private String body;

    @Column(name = "created", nullable = false)
    @CreationTimestamp
    private Timestamp created;

    @Column(name = "modified")
    @UpdateTimestamp
    private Timestamp modified;

    @ManyToOne(optional = false)
    @JoinColumn(nullable = false)
    private User author;

    @JsonIgnore
    @ManyToOne(optional = false)
    @JoinColumn(nullable = false)
    private Game game;

    protected Comment() {
    }

    public Comment(String body, User author, Game game) {
        this.body = body;
        this.author = author;
        this.game = game;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public Comment body(String body) {
        this.body = body;
        return this;
    }

    public Timestamp getCreated() {
        return created;
    }

    public void setCreated(Timestamp created) {
        this.created = created;
    }

    public Comment created(Timestamp created) {
        this.created = created;
        return this;
    }

    public Timestamp getModified() {
        return modified;
    }

    public void setModified(Timestamp modified) {
        this.modified = modified;
    }

    public Comment modified(Timestamp modified) {
        this.modified = modified;
        return this;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User user) {
        this.author = user;
    }

    public Comment author(User user) {
        this.author = user;
        return this;
    }

    public Game getGame() {
        return game;
    }

    public void setGame(Game game) {
        this.game = game;
    }

    public Comment game(Game game) {
        this.game = game;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Comment comment = (Comment) o;
        if (comment.id == null || id == null) {
            return false;
        }
        return Objects.equals(id, comment.id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }

    @Override
    public String toString() {
        return "Comment{" +
                "id=" + id +
                ", body='" + body + "'" +
                ", created='" + created + "'" +
                ", modified='" + modified + "'" +
                '}';
    }
}

File: cwt-spring/src/main/java/com/cwtsite/cwt/entity/Comment.kt
package com.cwtsite.cwt.entity

import com.cwtsite.cwt.domain.game.entity.Game
import com.cwtsite.cwt.domain.user.repository.entity.User
import com.fasterxml.jackson.annotation.JsonIgnore
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.util.Objects
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.SequenceGenerator
import javax.persistence.Table
import javax.validation.constraints.NotNull

@Entity
@Table(name = "comment")
@SequenceGenerator(name = "comment_seq", sequenceName = "comment_id_seq", allocationSize = 1)
data class Comment(

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "comment_seq")
    var id: Long? = null,

    @NotNull
    @Column(name = "body", nullable = false, columnDefinition = "text")
    var body: String? = null,

    @Column(name = "created", nullable = false)
    @field:CreationTimestamp
    var created: Instant? = null,

    @Column(name = "modified")
    @field:UpdateTimestamp
    var modified: Instant? = null,

    @ManyToOne(optional = false)
    @JoinColumn(nullable = false)
    var author: User? = null,

    @JsonIgnore
    @ManyToOne(optional = false)
    @JoinColumn(nullable = false)
    var game: Game? = null
) {

    constructor(body: String, author: User, game: Game) : this(id = null, body = body, author = author, game = game)

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val comment = other as Comment?
        return if (comment!!.id == null || id == null) {
            false
        } else id == comment.id
    }

    override fun hashCode(): Int {
        return Objects.hashCode(id)
    }

    override fun toString(): String {
        return "Comment{" +
            "id=" + id +
            ", body='" + body + "'" +
            ", created='" + created + "'" +
            ", modified='" + modified + "'" +
            '}'.toString()
    }
}
